pkg/cluster/cluster.go

```diff
@@ -211,11 +211,10 @@ func (c *Cluster) run(stopC <-chan struct{}, wg *sync.WaitGroup) {
 		if needDeleteCluster {
 			c.logger.Infof("deleting cluster")
 			c.delete()
+			go c.reportFailedStatus()
```

Comment:
I totally understand why the current behavior is wrong. but the root problem is the logical error.

moving reportFailedStatus into a bad designed condition will not make things better from code logic perspective.